docs/accounts/metrics_parameters.md

```diff
@@ -0,0 +1,42 @@
+
+For example, [about:welcome](about:welcome) self-hosts the form that captures the user's email address, then hands it off to Firefox Accounts servers to (among other things) determine if there is already an account associated with that address.
+
+**If you plan to go a similar route and self-host your own form, we ask that you do the following so that we can properly track top-of-funnel metrics** (such as volume of form views) associated with the page that hosts your FxA entrypoint:
+
+1. When the page that hosts your FxA entrypoint loads, have it make an XHR call to https://accounts.firefox.com/metrics-flow. The domain name of the request should match the FxA page that is being redirected to (e.g. https://accounts.firefox.com). You can use `fetch` to get this info.
+2. Include the following query parameters in the above request (see chart below for descriptions):
+  * `entrypoint`
+  * `utm_source`
+  * `utm_campaign`
+  * `form_type`
+  * An example: `https://accounts.firefox.com/metrics-flow?entrypoint=my_page&utm_source=my_referrer&utm_campaign=my_campaign&form_type=email`
+3. The response to metrics-flow will be a JSON object that contains the fields `flowId` and `flowBeginTime`. These values will need to be propagated to FxA as query parameters, which can be done using hidden form fields with the names `flow_id` and `flow_begin_time`. You can see an example of how the activity-stream team did this by looking [here](https://hg.mozilla.org/releases/mozilla-beta/diff/5d6261b568c6/browser/extensions/activity-stream/content-src/components/StartupOverlay/StartupOverlay.jsx#l1.22) (link will have to be updated).
+
+## Description of Metrics-Related Query Parameters
+**Note these may not be all the parameters you need to pass for your integration to work!** A more exhaustive but [less detailed list can be found here.](https://github.com/mozilla/fxa-content-server/blob/549fc459b851088ea910da182e17e748fa157f26/docs/query-params.md#context) What is documented below are only those that are relevant for metrics analysis in (e.g.) amplitude.
+
+Notes:
+* The "Regex Used for Validation" column shows the regular expression that the value of each parameter must conform to in order for the measurement to register in our metrics pipeline. If the value does not conform, all events associated with that parameter will fail to be logged.
+
+* You must have access to the mozilla amplitude account to see the example charts. If you are a Mozilla employee, you can email X at mozilla dot org for more information on gaining access to amplitude.
+
+|Name|Description|Example Values|Regex Used for Validation|Amplitude Chart Example|
+|----|-----------|-------------|------------------|-----------------|
+|`entrypoint`|The specific page or browser UI element containing the first step of the FxA sign-in/sign-up process (e.g., enter email form)|`firstrun`| /^[\w.-]+$/|[Firstrun form views](https://analytics.amplitude.com/mozilla-corp/chart/n8cd9no)|
```

Comment:
I predict the `Regex Used for Validation` column of this table will go stale and become a lie about what we accept. Unfortunately, there's approximately zero per cent chance of devs in the future remembering to come and update this document when they update the code.

I have 2 suggestions to fix this, one easy and one a little bit harder:

* We could just remove that column and provide a link to the source module that defines the regexes. Okay, looking at source code is a big ask for readers but we're already asking them to *understand regexes*, which is an even bigger ask than reading source code tbh. If they can understand a regex, they can read it from the source I think.

* Instead, we could keep the definitions here, but we could commit to making (and keeping) them machine-parseable from this document (e.g. we could wrap them in some kind of specially-formatted HTML comment that identifies the parameter they validate). If they're machine-parseable, we can write tests in the content server that fail if the definition here ever becomes a lie. Then it is impossible for future devs to forget to update these docs. (unless they are also in the habit of deleting failing tests of course, but we've got bigger problems if that's a thing!)

Reply:
(and if we are keeping the regexes in this table, let's wrap them in backticks too)

Reply:
>  Unfortunately, there's approximately zero per cent chance of devs in the future remembering to come and update this document when they update the code.

@philbooth speaks the truth.

>  (unless they are also in the habit of deleting failing tests of course, but we've got bigger problems if that's a thing!)

Ahhh... man... I thought this was a totally legit strategy for dealing with failing tests too.

Reply:
Can we leave a comment in the code to say to update this document if they change the regex?

Reply:
> Can we leave a comment in the code to say to update this document if they change the regex?

We could do that, but honestly I don't think it will make much difference. If linking to the source is not an option then I think it would be much better to spend a bit of dev time implementing my second suggestion, and write some automated tests that parse the regexes from this doc and fail whenever reality diverges from the documentation.

Reply:
(I'm happy to volunteer for that btw)

Reply:
If we are going to go through the bother of parsing the doc and checking the regexps against the validation code, could the script update the doc automatically and open a PR?

Reply:
It could do that, yep. Means messing about with GitHub API plus managing an API key for the script, so I'm probably not volunteering to do that in a first cut.

Reply:
I opted to remove the regex column for now, adding a link to the relevant code and a warning at the top of the chart. However, I like the idea of a failing test if the rules change. @philbooth it maybe its something I could do? depends on if a short overview of the right approach takes you less time than actually doing it (i know enough html, at least, so you may just need to to give me some hints re: the test logic). so its not a blocker for this PR, we can merge this without the tests then do that in a different PR.